bundles/core/src/test/java/com/adobe/cq/commerce/core/components/testing/Utils.java

```diff
@@ -82,6 +82,28 @@ public static String setupHttpResponse(String filename, HttpClient httpClient, i
      * @throws IOException
      */
     public static String setupHttpResponse(String filename, HttpClient httpClient, int httpCode, String startsWith) throws IOException {
+        return setupHttpResponse(filename, httpClient, httpCode, startsWith, null);
+    }
+
+    /**
+     * This method prepares the mock http response with either the content of the <code>filename</code>
+     * or the provided <code>content</code> String.<br>
+     * <br>
+     * <b>Important</b>: because of the way the content of an HTTP response is consumed, this method MUST be called each time
+     * the client is called.
+     *
+     * @param filename The file to use for the json response.
+     * @param httpClient The HTTP client for which we want to mock responses.
+     * @param httpCode The http code that the mocked response will return.
+     * @param startsWith When set, the body of the GraphQL POST request must start with that String.
+     * @param contains When set, the body of the GraphQL POST request must contain that String.
+     *
+     * @return The JSON content of that file.
+     *
+     * @throws IOException
+     */
+    public static String setupHttpResponse(String filename, HttpClient httpClient, int httpCode, String startsWith, String contains)
```

Comment:
I'm not entirely sure, but you don't need that at all, no? If not, please revert this change so we don't introduce something that we never use.

Reply:
See the above comment